src/SeaportBase.sol

```diff
@@ -0,0 +1,100 @@
+// SPDX-License-Identifier: BUSL-1.1
+pragma solidity 0.8.24;
+
+import { IERC20 } from "openzeppelin-contracts/contracts/interfaces/IERC20.sol";
+import { IIonPool } from "./interfaces/IIonPool.sol";
+import { IGemJoin } from "./interfaces/IGemJoin.sol";
+import { SeaportInterface } from "seaport-types/src/interfaces/SeaportInterface.sol";
+import { ItemType, OrderType } from "seaport-types/src/lib/ConsiderationEnums.sol";
+import { OrderParameters } from "seaport-types/src/lib/ConsiderationStructs.sol";
+
+contract SeaportBase {
+    error InvalidContractConfigs(IIonPool pool, IGemJoin join);
+
+    // Callback
+    error NotACallback();
+    error MsgSenderMustBeSeaport(address msgSender);
+
+    // Order parameters head validation
+    error OffersLengthMustBeOne(uint256 length);
+    error ConsiderationsLengthMustBeTwo(uint256 length);
+    error ZoneMustBeThis(address zone);
+    error OrderTypeMustBeFullRestricted(OrderType orderType);
+    error ZoneHashMustBeZero(bytes32 zoneHash);
+    error ConduitKeyMustBeZero(bytes32 conduitKey);
+    error InvalidTotalOriginalConsiderationItems();
+
+    // Offer item validation
+    error OItemTypeMustBeERC20(ItemType itemType);
+
+    // Consideration item 1 validation
+    error C1TypeMustBeERC20(ItemType itemType);
+    error C1TokenMustBeThis(address token);
+    error C1RecipientMustBeSender(address invalidRecipient);
+
+    // Consideration item 2 validation
+    error C2TypeMustBeERC20(ItemType itemType);
+
+    uint256 internal constant TSLOT_AWAIT_CALLBACK = 0;
+    uint256 internal constant TSLOT_COLLATERAL_DELTA = 1;
+
+    SeaportInterface public constant SEAPORT = SeaportInterface(0x00000000000000ADc04C56Bf30aC9d3c0aAF14dC);
+
+    uint8 public immutable ILK_INDEX;
+
+    IIonPool public immutable POOL;
+    IGemJoin public immutable JOIN;
+
+    IERC20 public immutable BASE;
+    IERC20 public immutable COLLATERAL;
+
+    /**
+     * @dev Modifier to check that the callback can only be called when initiated by
+     * this contract. Other contract calls that tries to enter this contract's callback
+     * will revert.
+     */
+    modifier onlyReentrant() {
+        uint256 deleverageInitiated;
+
+        assembly {
+            deleverageInitiated := tload(TSLOT_AWAIT_CALLBACK)
+        }
+
+        if (deleverageInitiated == 0) revert NotACallback();
+        _;
+    }
+
+    /**
+     * @dev Modifier to check that the callback can only be called by the Seaport contract.
+     */
+    modifier onlySeaport() {
+        if (msg.sender != address(SEAPORT)) revert MsgSenderMustBeSeaport(msg.sender);
+        _;
+    }
+
+    constructor(IIonPool pool, IGemJoin gemJoin, uint8 ilkIndex) {
+        POOL = pool;
+        JOIN = gemJoin;
+
```

Comment:
Not a massive deal but failed deployments would use less gas if these went below the input validation.